src/db/mod.rs

```diff
@@ -360,22 +367,9 @@ impl Database {
         password: Option<&str>,
         keyfile: Option<&mut dyn std::io::Read>,
     ) -> Result<Vec<u8>, DatabaseSaveError> {
-        let key_elements = Database::get_key_elements(password, keyfile)?;
-
-        let encrypted_db = match self.header {
-            Header::KDB(_) => {
-                return Err(DatabaseSaveError::UnsupportedVersion.into());
-            }
-            Header::KDBX3(_) => {
-                return Err(DatabaseSaveError::UnsupportedVersion.into());
-            }
-            Header::KDBX4(_) => {
-                self.generate_ivs()?;
-                crate::format::kdbx4::dump(self, &key_elements)
-            }
-        };
-
-        encrypted_db
+        let mut data = Vec::new();
```

Comment:
@sseemayer ah! That's what I was looking for yesterday, I didn't realize that you could use a simple `Vec` as an `std::io::Write`! I don't think the `dump` function is relevant anymore in that case.